pkg/services/ppmshipment/ppm_estimator.go

```
@@ -644,16 +646,12 @@ func (f estimatePPM) priceBreakdown(appCtx appcontext.AppContext, ppmShipment *m
 	var pickupPostal, destPostal string
 
 	// Check different address values for a postal code
-	if ppmShipment.ActualPickupPostalCode != nil {
-		pickupPostal = *ppmShipment.ActualPickupPostalCode
-	} else if ppmShipment.PickupAddress.PostalCode != "" {
+	if ppmShipment.PickupAddress != nil && ppmShipment.PickupAddress.PostalCode != "" {
 		pickupPostal = ppmShipment.PickupAddress.PostalCode
 	}
 
 	// Same for destination
-	if ppmShipment.ActualDestinationPostalCode != nil {
-		destPostal = *ppmShipment.ActualDestinationPostalCode
-	} else if ppmShipment.DestinationAddress.PostalCode != "" {
+	if ppmShipment.DestinationAddress != nil && ppmShipment.DestinationAddress.PostalCode != "" {
 		destPostal = ppmShipment.DestinationAddress.PostalCode
 	}
```

Should we return an error here since we technically don't have the data to calculate?

---

I think this is out of scope of this work, going to leave this as is

---

This just seems like basic error handling and coding practices - I don't agree that it's "out of scope" tbh

---

To me whats in scope of this work is to maintain current functionality in the app and remove references to the actual postal code columns. We previously were using the initialized zero value of `pickupPostal, destPostal` if `ppmShipment.DestinationAddress.PostalCode != ""` or `ppmShipment.PickupAddress.PostalCode != ""` so I think maintaining that behavior is the best path.

Happy to continue discussing though.

---

I understand. If you are dead set on leaving it, I'll resolve this but will remove myself as a reviewer as I don't want to approve something I don't particularly agree with. All good though!

---

All good here too, thanks for talking it out, and understand you removing yourself as a reviewer 💯 .

I get where you're coming from as well - I'll mull it over a bit more and bring up at team sync. Thanks as always for discussion

---

I still don't understand what there is to mull over.

If the destination address is nil, then we do not have the information to get the distance which factors in with calculating the incentive. If you don't return an error here then we'll get a nil pointer error later on in the function.

---

The result would still be the same

---

`if len(pickupPostal) >= 3 && len(destPostal) >= 3 && pickupPostal[:3] == destPostal[:3] {`

This will not result in an error - pickupPostal and destPostal will both be of length 0.

---

True, but we still need the logic to be done correctly

---

UPDATED